pyo3-derive-backend/src/py_method.rs

```diff
@@ -10,23 +10,80 @@ use utils;
 pub fn gen_py_method<'a>(
     cls: &syn::Type,
     name: &syn::Ident,
-    sig: &mut syn::MethodSig,
-    meth_attrs: &mut Vec<syn::Attribute>,
+    sig: &syn::MethodSig,
+    meth_attrs: &Vec<syn::Attribute>,
+    spec: &FnSpec,
 ) -> TokenStream {
     check_generic(name, sig);
 
     let doc = utils::get_doc(&meth_attrs, true);
-    let spec = FnSpec::parse(name, sig, meth_attrs);
+
+    macro_rules! make_py_method_def {
```

Comment:
Could this also be a normal function?

Reply:
Yes, but it would require 2 more arguments passed to each invocation of the macro/function (the local macro has access to the locals `doc` and `name` without passing them explicitly) and wrapping the fourth argument into a lambda. Do you think this is preferable? I don't really have a preference here.